subsys/pm/device_runtime.c

```diff
@@ -152,11 +157,13 @@ int pm_device_runtime_get(const struct device *dev)
 
 	pm->usage++;
 
-	if (!k_is_pre_kernel()) {
+	if (!k_is_pre_kernel() && pm->state == PM_DEVICE_STATE_SUSPENDING) {
+		k_mutex_unlock(&pm->lock);
+
 		/* wait until possible async suspend is completed */
-		while (pm->state == PM_DEVICE_STATE_SUSPENDING) {
-			(void)k_condvar_wait(&pm->condvar, &pm->lock, K_FOREVER);
-		}
+		k_event_wait(&pm->event, EVENT_MASK, true, K_FOREVER);
+
+		(void)k_mutex_lock(&pm->lock, K_FOREVER);
```

Comment:
shouldn't we keep the loop here? ie

```
while (pm->state == PM_DEVICE_STATE_SUSPENDING) {
	k_mutex_unlock(&pm->lock);
	/* wait until possible async suspend is completed */
	k_event_wait(&pm->event, EVENT_MASK, true, K_FOREVER);
	(void)k_mutex_lock(&pm->lock, K_FOREVER);
}
```

I may be preempted just after k_event_wait, and return to this point with PM_DEVICE_STATE_SUSPENDING set again (theoretically, unlikely though).